### PR TITLE
Review

### PR DESCRIPTION
**Pull request by EmirVatric**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      const newLetter = getLetterFromOrder(newLetterOrder, isCapital);
+      result.push(newLetter);
+    }
+  }
+  return result.join('');
+};
+
+
+const arrayAnalysis = (array) => {
+  const average = Math.floor((array.reduce((a, b) => a + b)) / array.length);
```
Comment (BalintMendli) [inline index.js]: Why do we need `Math.floor` here?

File: index.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+test('It returns smallest number in array', () => {
+  expect(arrayAnalysis([1, 2, 3]).min).toBe(1);
+});
+
+test('It returns biggest number in array', () => {
+  expect(arrayAnalysis([1, 2, 3]).max).toBe(3);
+});
+
+test('It returns length of the array', () => {
+  expect(arrayAnalysis([1, 2, 3]).length).toBe(2);
```
Comment (BalintMendli) [inline index.test.js]: This should be 3. We are mutating the input array so it gives us incorrect result.

**Review by BalintMendli — CHANGES_REQUESTED** [review]
Hey @EmirVatric @abdellani ,

Great work on the tests. 👏 

Everything looks good, we just have one issue. The arrayAnalysis function doesn't work correctly. Please fix it.

[Optional] Split up the functions and tests into separate files and use ES6 modules.

Please update the project and submit another review request.

**Review by JamesOkunlade — APPROVED** [review]
### Nice job 👏 It's a 👍 for me.

**Please go ahead and merge.**

**Status: `Approved`**

**Good luck with your future projects. 🤞**

**Pull request merged**